### PR TITLE
Fixes #16332 (slurm plugin - mlflow logger compatibility) 

### LINKED ISSUE #16332: SLURM signal and mlflow logger not compatible
### Bug description

I have a mlflow logger coupled with a slurm plugin and at the end of the slurm job I have a bug in the mlflow logger due to a bad string that should be in upper case.

### How to reproduce the bug

```python
mlf_logger = MLFlowLogger(experiment_name="XXXXX", tracking_uri=PATH_MLFLOW)

    if MODE == "training":
        modele.train()
        trainer = pl.Trainer(max_time={"hours": NB_HOURS}, logger=mlf_logger,
                                ... , plugins=[SLURMEnvironment(requeue_signal=signal.SIGUSR1)])
```

### Error messages and logs

```
File "/opt/conda/lib/python3.7/site-packages/pytorch_lightning/trainer/connectors/signal_connector.py", line 33, in __call__ signal_handler(signum, frame)
File "/opt/conda/lib/python3.7/site-packages/pytorch_lightning/trainer/connectors/signal_connector.py", line 74, in slurm_sigusr_handler_fn logger.finalize("finished")  
File "/opt/conda/lib/python3.7/site-packages/lightning_utilities/core/rank_zero.py", line 24, in wrapped_fn
return fn(*args, **kwargs)
File "/opt/conda/lib/python3.7/site-packages/pytorch_lightning/loggers/mlflow.py", line 265, in finalize 
self.experiment.set_terminated(self.run_id, status)
File "/opt/conda/lib/python3.7/site-packages/mlflow/tracking/client.py", line 1647, in set_terminated self._tracking_client.set_terminated(run_id, status, end_time)
File "/opt/conda/lib/python3.7/site-packages/mlflow/tracking/_tracking_service/client.py", line 508, in set_terminated    run_status=RunStatus.from_string(status),
File "/opt/conda/lib/python3.7/site-packages/mlflow/entities/run_status.py", line 22, in from_string "status strings: %s" % (status_str, list(RunStatus._STRING_TO_STATUS.keys()))  
Exception: Could not get run status corresponding to string finished. Valid run status strings: ['RUNNING', 'SCHEDULED', 'FINISHED', 'FAILED', 'KILLED']                            
srun: error: cluster: task 0: Exited with exit code 1     

```

### Environment

<details>
  <summary>Current environment</summary>

```
#- Lightning Component (e.g. Trainer, LightningModule, LightningApp, LightningWork, LightningFlow):
#- PyTorch Lightning Version (e.g., 1.5.0):
#- Lightning App Version (e.g., 0.5.2):
#- PyTorch Version (e.g., 1.10):
#- Python version (e.g., 3.9):
#- OS (e.g., Linux):
#- CUDA/cuDNN version:
#- GPU models and configuration:
#- How you installed Lightning(`conda`, `pip`, source):
#- Running environment of LightningApp (e.g. local, cloud):
```

</details>

### More info

When I look at the signal function:
```python
    def slurm_sigusr_handler_fn(self, signum: _SIGNUM, frame: FrameType) -> None:
        rank_zero_info("handling auto-requeue signal")

        # save logger to make sure we get all the metrics
        for logger in self.trainer.loggers:
            logger.finalize("finished")
```
The problem seems to be that "finished" is in lowcase but mlflow need an upper case ...
I can push a simple modification in the mlflow logger to change the "finished" in upper case.

### PR DESCRIPTION
**Pull request by Forbu**
## What does this PR do?

A very simple modification of the MLFlowLogger class to handle the slurm plugin.

Fixes #16332 

### Does your PR introduce any breaking changes?

Minimal change.
MLflow accept ['RUNNING', 'SCHEDULED', 'FINISHED', 'FAILED', 'KILLED'] as a list a status so nothing can break here.

## Before submitting

- [ X] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ X] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ X] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃
Well looking at lightning code was fun ^^ it is my first real pull request on an open source repo :) (although very simple PR).

### CODE REVIEW

**Review by awaelchli — APPROVED** [review]
Thanks for the quick fix @Forbu

**Review by carmocca — APPROVED** [review]
